packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -252,6 +253,23 @@ function s(
                         setVMBeingRendered(vmBeingRenderedInception);
                     }
                 } else {
+                    if (isVBaseElement(vnode)) {
+                        const { renderMode } = getVMBeingRendered()!;
+                        if (renderMode === RenderMode.Light) {
+                            // Note, light DOM slots directly return the children of the slotted content and skips creating
+                            // a slot vnode/element.
+                            // In the absence of a slot element, we replace the slot attribute on the element with the one identified
+                            // on the slot element in order to preserve the slot mapping across nested slots.
+                            // ex: <slot slot="slotName"></slot>, data.attrs.slot => "slotName"
+                            const slotName = data.attrs?.slot;
+                            if (vnode.data.attrs?.slot !== slotName) {
+                                vnode.data = {
+                                    ...vnode.data,
+                                    attrs: { ...vnode.data.attrs, slot: slotName },
+                                };
```

Comment:
Creating a copy instead of overwriting the attributes here since everything in `VElementData` should be read only.

Reply:
Might be worth adding a reminder to the comment here that `VNodeData` might be shared between `VNodes` as an optimization, which is why `vnode.data` is unsafe to mutate.